src/scripts/main.js

```diff
@@ -1,3 +1,44 @@
 'use strict';
 
-// write code here
+function sortRowsByNumber(rows, childSelector) {
+  let result = [];
+  const childNomer = childSelector + 1;
+
+  if (childSelector === 0 || childSelector === 1) {
+    result = [...rows].sort((a, b) => {
+      const lettersA = a.querySelector(
+        `td:nth-child(${childNomer})`,
+      ).textContent;
+      const lettersB = b.querySelector(
+        `td:nth-child(${childNomer})`,
+      ).textContent;
+
+      return lettersB.localeCompare(lettersA);
+    });
+  } else {
+    result = [...rows].sort((a, b) => {
+      const salaryA = +a
+        .querySelector(`td:nth-child(${childNomer})`)
+        .textContent.replace(/,/g, '')
+        .replace(/\$/, '');
+      const salaryB = +b
+        .querySelector(`td:nth-child(${childNomer})`)
+        .textContent.replace(/,/g, '')
+        .replace(/\$/, '');
+
+      return salaryA - salaryB;
+    });
+  }
+
+  const tbody = document.querySelector('tbody');
+
+  tbody.innerHTML = '';
+  result.forEach((row) => tbody.appendChild(row));
+}
+
+const tableRows = document.querySelectorAll('tbody tr');
+const filter = document.querySelectorAll('thead tr th');
```

Comment:
The variable 'filter' might be better named as 'headers' or 'tableHeaders' to more accurately describe its purpose.